docs/tutorials/typescript/nextjs/index.md

```diff
@@ -1,290 +1,626 @@
+Now that you know the API works, you can build the user interface.
 
-## Deploying your Temporal + Next.js app
+## Build the front-end interface
 
-Your Next.js app, including Next.js API Routes with Temporal Clients in them, can be deployed anywhere Next.js can be deployed, including in serverless environments like Vercel or Netlify.
+You can call your Next.js API with `curl`, but that's not the idea user interface.
+You'll use React components with Next.js to make a request to the API you just created to create the one-click buy experience.
 
-:::important
+To call the API Route from the Next.js frontend, you'll use the `fetch` API to make a request o the `/api/startbuy` method using a similar payload when you click a button.
 
-However, your Temporal Workers **must** be deployed in traditional "serverful" environments (e.g. with EC2, Digital Ocean or Render, not a serverless environment).
+Build out the front-end user interface. Open `app/page.tsx` and remove the contents of the file that the project generator created.
+Then at the top, add the following code to add directives and import the necessary libraries:
 
-:::
+<!--SNIPSTART typescript-next-oneclick-page-start -->
+[app/page.tsx](https://github.com/temporalio/nextjs-temporal-one-click-template/blob/next-v2/app/page.tsx)
+```tsx
+'use client'
+import Head from 'next/head';
+import React from 'react';
+import { v4 as uuid4 } from 'uuid';
+import 'react-toastify/dist/ReactToastify.css';
+```
+<!--SNIPEND-->
 
-**Both Temporal Clients and Temporal Workers must be configured to communicate with a Temporal Server instance**, whether self-hosted or Temporal Cloud.
-You will need to configure gRPC connection address, namespace, and mTLS cert and key (strongly recommended).
+Next, define a TypeScript Interface for the Product's properties, a Type to define states for the purchase, and a collection of Products:
+
+<!--SNIPSTART typescript-next-oneclick-page-vars -->
+[app/page.tsx](https://github.com/temporalio/nextjs-temporal-one-click-template/blob/next-v2/app/page.tsx)
+```tsx
+
+interface ProductProps {
+  product: {
+    id: number;
+    name: string;
+    price: string;
+  };
+}
+
+const products = [
+  {
+    id: 1,
+    name: 'PDF Book',
+    price: '$49',
+  },
+  {
+    id: 2,
+    name: 'Kindle Book',
+    price: '$49',
+  },
+];
+
+type ITEMSTATE = 'NEW' |  'ORDERED' | 'ERROR';
+```
+<!--SNIPEND-->
+
+In a production application, your products would come from another part of your application, such as a database or a Shopify store.
+
+Next, define a `Product` component with the following code:
+
+<!--SNIPSTART typescript-next-oneclick-page-product -->
+[app/page.tsx](https://github.com/temporalio/nextjs-temporal-one-click-template/blob/next-v2/app/page.tsx)
+```tsx
+const Product: React.FC<ProductProps> = ({ product }) => {
+  const itemId = product.id;
+  const [state, setState] = React.useState<ITEMSTATE>('NEW');
```

Comment:
Slight nit but can just import useState, useRef, useCallback if you wanted to, but nbd at all. IE:

```
import React, { useState, useRef  } from 'react';

const [state, setState] = useState<ITEMSTATE>('NEW');
```